OptimizelySDK.Tests/OptimizelyJsonTest.cs

```diff
@@ -0,0 +1,225 @@
+            Assert.AreEqual(optimizelyJSONUsingMap.GetValue<bool>("boolField"), true);
+            Assert.AreEqual(optimizelyJSONUsingMap.GetValue<int>("objectField.inner_field_int"), 3);
+            Assert.AreEqual(optimizelyJSONUsingMap.GetValue<double>("objectField.inner_field_double"), 13.21);
+            Assert.AreEqual(optimizelyJSONUsingMap.GetValue<string>("objectField.inner_field_string"), "john");
+            Assert.AreEqual(optimizelyJSONUsingMap.GetValue<bool>("objectField.inner_field_boolean"), true);
+            Assert.IsTrue(optimizelyJSONUsingMap.GetValue<Dictionary<string, object>>("objectField") is Dictionary<string, object>);
+        }
+
+        [Test]
+        public void TestOptimizelyJsonGetVariablesWhenSetUsingString()
+        {
+            var optimizelyJSONUsingString = new OptimizelyJson(Payload, ErrorHandlerMock.Object, LoggerMock.Object);
+
+            Assert.AreEqual(optimizelyJSONUsingString.GetValue<long>("field1"), 1);
+            Assert.AreEqual(optimizelyJSONUsingString.GetValue<double>("field2"), 2.5);
+            Assert.AreEqual(optimizelyJSONUsingString.GetValue<string>("field3"), "three");
+            Assert.AreEqual(optimizelyJSONUsingString.GetValue<long>("field4.inner_field1"), 3);
+            Assert.True(TestData.CompareObjects(optimizelyJSONUsingString.GetValue<List<object>>("field4.inner_field2"), new List<object>() { "1", "2", 3, 4.23, true }));
+        }
+
+        [Test]
+        public void TestGetValueReturnsEntireDictWhenJsonPathIsEmptyAndTypeIsValid()
+        {
+            var optimizelyJSONUsingString = new OptimizelyJson(Payload, ErrorHandlerMock.Object, LoggerMock.Object);
+            var actualDict = optimizelyJSONUsingString.ToDictionary();
+            var expectedValue = optimizelyJSONUsingString.GetValue<Dictionary<string, object>>("");
+            Assert.NotNull(expectedValue);
+            Assert.True(TestData.CompareObjects(expectedValue, actualDict));
+        }
+
+        [Test]
+        public void TestGetValueReturnsDefaultValueWhenJsonIsInvalid()
+        {
+            var payload = "{ \"field1\" : {1:\"Csharp\", 2:\"Java\"} }";
+            var optimizelyJSONUsingString = new OptimizelyJson(payload, ErrorHandlerMock.Object, LoggerMock.Object);
+            var expectedValue = optimizelyJSONUsingString.GetValue<Dictionary<float, string>>("field1");
+            // Even though above given JSON is not valid, newtonsoft is parsing it so
+            Assert.IsNotNull(expectedValue);
+        }
+
+        [Test]
+        public void TestGetValueReturnsDefaultValueWhenTypeIsInvalid()
+        {
+            var payload = "{ \"field1\" : {\"1\":\"Csharp\",\"2\":\"Java\"} }";
+            var optimizelyJSONUsingString = new OptimizelyJson(payload, ErrorHandlerMock.Object, LoggerMock.Object);
+            var expectedValue = optimizelyJSONUsingString.GetValue<Dictionary<float, string>>("field1");
```

Comment:
shouldn't it be <string, string> instead of <float, string> ?

Reply:
No, Actually it's to test that when invalid type is passed in getValue it will return  default value.